core/templates/dev/head/pages/exploration_player/TutorCardDirective.js

```diff
@@ -100,6 +100,8 @@ oppia.directive('tutorCard', [
 
             AudioPlayerService.stop();
             AutogeneratedAudioPlayerService.cancel();
+
+            oppiaPlayerService.recordStateFinished(index);
```

Comment:
I don't think this is quite the right place for this. I think the place to call the event is just after the answer has been evaluated as falling in an answer group that leads to a different state.

Reply:
@seanlip Again, this falls back to our question on what we mean by an active card (I'll be running some checks to figure that out). We want num_completions to be the number of times the state was completed (and not count number of times the state looped the learner back to another state which would eventually lead back to this one).

I think I might be  misunderstanding the meaning of active card. If a learner submits an answer to some state B and gets looped back to A, I was of the impression that state B is still the active state. If this understanding is wrong, we again have to rethink how we plan to implement num_completions (whole idea of num_completions was: quitters_count = first_hit_count - num_completions, multiple recording of state finishes won't be helping us achieve that).

Reply:
I'm pretty sure that that's not how it works, yeah. Definitely please do double check, but I don't think the active card is state B in the scenario you are describing.

For num_completions I guess you could do quitters_count = total_state_completions - total_state_starts for a given card (counting with multiplicity)?

Reply:
WE're counting with multiplicity now. No issues anymore and we can record state completions now wherever we are recording state transitions. Fixed.